src/e2e/puppeteer/__tests__/caret-multiline.ts

```diff
@@ -0,0 +1,266 @@
+    const importText = `
+  - a
+    - b
+      - c
+      - Beautiful antique furnishings fill this quiet, comfortable flat across from the Acropolis museum. AC works great. It is in an heavily touristic area, but the convenience can't be beat. Highly recommended.
+      - d`
+
+    await paste(importText)
+
+    const editableNodeHandle = await waitForEditable('b')
+    await click(editableNodeHandle, { edge: 'right' })
+
+    await press('ArrowUp')
+
+    // the focus must be at the beginning of 'a' after cursor up
+    const textContext = await getSelection().focusNode?.textContent
+    expect(textContext).toBe('a')
+
+    const offset = await getSelection().focusOffset
+    expect(offset).toBe(0)
+  })
+
+  // test case 5
+  it.skip('on 2x cursorDown, the caret should move from the end of a cursor into the 2nd line of the new multi-line cursor.', async () => {
+    const multiLineCursor =
+      "Beautiful antique furnishings fill this quiet, comfortable flat across from the Acropolis museum. AC works great. It is in an heavily touristic area, but the convenience can't be beat. Highly recommended."
+    const importText = `
+  - a
+    - b
+      - c
+      - ${multiLineCursor}
+      - d`
+
+    await paste(importText)
+
+    const editableNodeHandle = await waitForEditable('c')
+    await click(editableNodeHandle, { edge: 'left' })
+
+    await press('ArrowDown')
+    await press('ArrowDown')
+
+    // the focus must be in the middle of the multi-line thought after cursor down
+    const textContext = await getSelection().focusNode?.textContent
+    expect(textContext).toBe(multiLineCursor)
+
+    const offset = await getSelection().focusOffset
+    expect(offset).toBeGreaterThan(multiLineCursor.length / 3)
+    expect(offset).toBeLessThan((multiLineCursor.length * 2) / 3)
+  })
+
+  // test case 6
+  it('on cursorDown, the caret should move from the beginning of a multi-line cursor into the 2nd line of the same cursor.', async () => {
+    const multiLineCursor =
+      "Beautiful antique furnishings fill this quiet, comfortable flat across from the Acropolis museum. AC works great. It is in an heavily touristic area, but the convenience can't be beat. Highly recommended."
+    const importText = `
+  - a
+    - b
+      - c
+      - ${multiLineCursor}
+      - d`
+
+    await paste(importText)
+
+    const editableNodeHandle = await waitForEditable(multiLineCursor)
+    await click(editableNodeHandle, { offset: 2 })
+
+    await press('ArrowDown')
+
+    // the focus must be in the middle of the multi-line cursor after cursor down
+    const textContext = await getSelection().focusNode?.textContent
+    expect(textContext).toBe(multiLineCursor)
+
+    const offset = await getSelection().focusOffset
+    expect(offset).toBeGreaterThan(multiLineCursor.length / 3)
+    expect(offset).toBeLessThan((multiLineCursor.length * 2) / 3)
```

Comment:
I think this math is misleading, as it gives the impression that Line 1 corresponds to the first third of the string, Line 2 corresponds to the second third of the string, etc. In actuality that would only be true at a specific screen width, which we have not set. I don't want this test to have an implicit dependency on a puppeteer browser width that we have not explicitly set.

I'd like to suggest something along the lines of `offsetEnd > offsetStart && offsetEnd < multilineCursor.length`. This correctly asserts that the caret has moved to the right, but not to the very end, which by all accounts means it has moved down one line. It does this without making any assumptions about the length of the string or the width of the browser, i.e. where the line wraps.

If I am missing something or you have an alternative suggestions please let me know!

Reply:
I think you're right. It seemed like a clever solution, but it's still too fiddly.